storage/src/test/java/org/apache/kafka/server/log/remote/metadata/storage/TopicBasedRemoteLogMetadataManagerTest.java

```diff
@@ -152,15 +152,17 @@ public void testNewPartitionUpdates() throws Exception {
 
         // RemoteLogSegmentMetadata events are already published, and topicBasedRlmm's consumer manager will start
         // fetching those events and build the cache.
-        waitUntilConsumerCatchesUp(newLeaderTopicIdPartition, newFollowerTopicIdPartition, 30_000L);
-
+        waitUntilConsumerCatchesUp(newLeaderTopicIdPartition, newFollowerTopicIdPartition, 30_000L, 0, 0);
         Assertions.assertTrue(topicBasedRlmm().listRemoteLogSegments(newLeaderTopicIdPartition).hasNext());
         Assertions.assertTrue(topicBasedRlmm().listRemoteLogSegments(newFollowerTopicIdPartition).hasNext());
     }
 
     private void waitUntilConsumerCatchesUp(TopicIdPartition newLeaderTopicIdPartition,
                                             TopicIdPartition newFollowerTopicIdPartition,
-                                            long timeoutMs) throws TimeoutException {
+                                            long timeoutMs,
+                                            long targetLeaderMetadataPartitionOffset,
```

Comment:
These parameters will not help much here as this method was written for `testNewPartitionUpdates` but other tests in this class used the functionality with the gaps. It is better to relook at those usecases and refactor this method respectively.